File: snippets/snippets/node/parse-v2-post/blocks.js
import { CohereClient } from 'cohere-ai';

const cohere = new CohereClient({});

(async () => {
  const response = await cohere.v2.parse({
    model: 'parse-v5.0',
    document: {
      type: 'image_url',
      imageUrl: 'https://cohere.com/favicon-32x32.png',
    },
    outputFormat: 'blocks',
  });
  console.log(response);
})();

File: snippets/snippets/java/app/src/main/java/parsev2post/Blocks.java
package parsev2post; /* (C)2024 */

import com.cohere.api.Cohere;
import com.cohere.api.resources.v2.requests.ParseRequest;
import com.cohere.api.types.ParseDocument;
import com.cohere.api.types.ParseOutputFormat;
import com.cohere.api.types.ParseResponse;

public class Blocks {
  public static void main(String[] args) {
    Cohere cohere = Cohere.builder().clientName("snippet").build();

    ParseResponse response =
        cohere
            .v2()
            .parse(
                ParseRequest.builder()
                    .model("parse-v5.0")
                    .document(
                        ParseDocument.builder()
                            .imageUrl("https://cohere.com/favicon-32x32.png")
                            .build())
                    .outputFormat(ParseOutputFormat.BLOCKS)
                    .build());

    System.out.println(response);
  }
}
